shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/widget/presentation/StudyPlanWidgetReducer.kt

```diff
@@ -32,13 +31,17 @@ class StudyPlanWidgetReducer : StateReducer<State, Message, Action> {
     override fun reduce(state: State, message: Message): StudyPlanWidgetReducerResult =
         when (message) {
             is Message.Initialize ->
-                coldContentFetch()
+                if (state.sectionsStatus == StudyPlanWidgetFeature.ContentStatus.IDLE ||
+                    state.sectionsStatus == StudyPlanWidgetFeature.ContentStatus.ERROR && message.forceUpdate
+                ) {
+                    coldContentFetch()
+                } else {
+                    null
+                }
```

Comment:
I think it's better to extract this message handling into a function, for example `handleInitializeMessage`, to make all code in the reducer in the same style

cc @XanderZhu